### ISSUE TITLE
usage of sigprocmask fails to compile when libc is linked

### ISSUE THREAD

**Issue by goshhhy** [ziglang/zig#14861]
### Zig Version

0.11.0-dev.1908+06b263825

### Steps to Reproduce and Observed Behavior

compile the following source code twice, once without libc linked, and once with:

```
const std = @import("std");

pub export fn sigclearmask() void {
    std.os.sigprocmask(std.os.SIG.SETMASK, &std.os.empty_sigset, null);
}
```

when linked without libc, the code compiles without an error.

```
linear@yon:~/code/zigtest $ zig build                                                                                                                                                                                                               
linear@yon:~/code/zigtest $ 
```

when linked with libc, the code produces the following:
```
linear@yon:~/code/zigtest $ zig build
/usr/lib/zig/std/os.zig:5546:38: error: expected type 'c_int', found 'u32'
    switch (errno(system.sigprocmask(flags, set, oldset))) {
                                     ^~~~~
/usr/lib/zig/std/os.zig:5546:38: note: signed 32-bit int cannot represent all possible unsigned 32-bit values
/usr/lib/zig/std/c.zig:84:44: note: parameter type declared here
        pub extern "c" fn sigprocmask(how: c_int, noalias set: ?*const c.sigset_t, noalias oset: ?*c.sigset_t) c_int;
                                           ^~~~~
referenced by:
    sigclearmask: src/main.zig:4:11
    remaining reference traces hidden; use '-freference-trace' to see all reference traces

error: zigtest...
error: The following command exited with error code 1:
/usr/bin/zig build-lib /home/linear/code/zigtest/src/main.zig -lc --cache-dir /home/linear/code/zigtest/zig-cache --global-cache-dir /home/linear/.cache/zig --name zigtest -static --enable-cache 
error: the following build command failed with exit code 1:
/home/linear/code/zigtest/zig-cache/o/d2f464da8f9ed38146bcdaec7e7d1f15/build /usr/bin/zig /home/linear/code/zigtest /home/linear/code/zigtest/zig-cache /home/linear/.cache/zig
linear@yon:~/code/zigtest $ 
```

### Expected Behavior

i expect the result to be the same regardless of whether or not libc is linked.

**Comment by goshhhy** [comment]
further analysis: std.os.sigprocmask appears to call out to std.os.system.sigprocmask, which is either std.os.linux.sigprocmask, or std.c.sigprocmask. 

these functions have incompatible definitions - std.os.linux.sigprocmask takes a u32 as its first arg, but std.c.sigprocmask takes a c_int.

**Comment by goshhhy** [comment]
closed as a duplicate of #13234
